vignettes/including-data-in-teal-applications.Rmd

```diff
@@ -0,0 +1,251 @@
+`teal_data` objects are passed to `init` to build the application, where they are modified by the filter panel (if applicable) and passed on to modules.
+Thus, the first step of building a `teal` app is creating a `teal_data` object.
+
+### General Data
+
+A `teal_data` object is created by calling the `teal_data` function and passing data objects as `name:value` pairs.
+
+```{r, message=FALSE}
+library(teal)
+
+# create teal_data
+data <- teal_data(iris = iris, cars = mtcars)
+```
+
+Note that `iris` and `cars` have been added to the `datanames` property of `data` (jump to [`teal_data` properties]).
+
+This is sufficient to run a `teal` app.
+
+```{r, eval=FALSE}
+# build app
+app <- init(
+  data = data,
+  modules = example_module()
+)
+
+# run app
+shinyApp(app$ui, app$server)
+```
+
+### Reproducible Data
+
+A `teal_data` stores data in a separate environment.
+Therefore, modifying the stored datasets requires that code be evaluated in that environment.
+Following that logic, one can create an empty `teal_data` object and populate it by evaluating code.
+This can be done using the `eval_code` function or, more conveniently, using the `within` function.
+
+```{r}
+# create empty object
+data0 <- teal_data()
+
+# run code in the object
+data1 <- teal.code::eval_code(data0, code = "iris <- iris
+                                             cars <- mtcars"
+)
+# alternative
+data2 <- within(data0, {
+  iris <- iris
+  cars <- mtcars
+})
+```
+
+The key difference between `eval_code` and `within` is that the former accepts code as character vector or language objects (calls and expressions), while `within` accepts _only_ inline code.
+See `?eval_code` for more details.
+
+Note that with `data` the code that created the data objects is unknown so the process cannot be reproduced.
+The necessary code can be added with the `code` argument of the `teal_data` function but in that case the code may not reproduce the environment.
+Such an object is considered _unverified_ (jump to [`teal_data` properties]).
+
+If reproducibility is required, we recommend creating `teal_data` empty and evaluating code.
+
+###### code from file
+
+The ability to pass code as character vector to `eval_code` opens the door to using code stored in a file.
+```{r, eval=FALSE}
+# not run
+data_from_file <- teal_data()
+data_from_file <- eval_code(data, readLines("<path_to_file>"))
+```
+
+### Creating Data In-App
+
+The one departure from passing a `teal_data` object to `init` is when the data does not exist in the environment where the app is run, _e.g._ when it has to be pulled from a remote source.
+In those cases a `teal_data_module` must be used.
+See [this vignette](data-as-shiny-module.html) for a detailed description.
+
+<br>
+
+## Clinical Data
+
+Currently `teal` supports two specialized data formats.
+
+### `ADaM` Data
+
+The container used for `CDISC` data is the `cdisc_data` class, which is an extension of `teal_data` created with the `cdisc_data` function.
```

Comment:
There is no `cdisc_data` class❗

Target for teal is to be data-standard independent and we are very close to achieve this. At this moment `teal.data` supports every kind of data. It is only `teal.slice` and `teal.transform` which really blocks us to announce this wonderful news ;]

```suggestion
For `CDISC` data we recommend to create `teal_data` object using `cdisc_data()` function, which adds default `join_keys` to set proper relationships with `ADSL` dataset.
```